src/poetry/utils/authenticator.py

```diff
@@ -463,6 +466,13 @@ def _get_certs_for_url(self, url: str) -> RepositoryCertificateConfig:
             return selected.certs(config=self._config)
         return RepositoryCertificateConfig()
 
+    def get_cached_file_for_url(self, url: str) -> Path | None:
+        if self._disable_cache:
```

Comment:
if you instead ask `if self._cache_control is None:` - which is also done elsewhere in this code - then the later cast will become unnecessary

(generally prefer to avoid casting.  If and when someone submits an MR that makes the cast incorrect, mypy has no way to say so.  So if there was not a simple way around this, would prefer to assert that self._cache_control is not None)

Reply:
Hadn't realized that `disable_cache` implies `self._cache_control is None`, although that's pretty obvious now I must admit :D thanks!